libqtile/bar.py

```diff
@@ -330,13 +339,19 @@ def _remove_crashed_widgets(self):
 
         for i in self.crashed_widgets:
             index = self.widgets.index(i)
-            crash = ConfigErrorWidget(widget=i)
-            crash._configure(self.qtile, self)
-            if self.horizontal:
-                crash.offsety = self.border_width[0]
-            else:
-                crash.offsetx = self.border_width[3]
-            self.widgets.insert(index, crash)
+
+            # Widgets that aren't available on the current backend should not
+            # be shown as "crashed" as the behaviour is expected. Only notify
+            # for genuine crashes.
+            if not i.supported_backends or (self.qtile.core.name in i.supported_backends):
```

Comment:
So widgets that are incompatible with the backend are just invisibly removed?

That's pretty nice - no need for backend checks or whatever in your config, just keep incompatible widgets defined and Qtile will deal with it.

Reply:
Exactly.